sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/DoubleHistogramPointData.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.data;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.common.Labels;
+import io.opentelemetry.sdk.metrics.common.ImmutableDoubleArray;
+import io.opentelemetry.sdk.metrics.common.ImmutableLongArray;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * DoubleHistogramPointData represents an approximate representation of the distribution of
+ * measurements.
+ */
+@Immutable
+@AutoValue
+public abstract class DoubleHistogramPointData implements PointData {
+  /**
+   * Functional interface for consuming bucket boundaries and counts as a sequence of pair values.
+   */
+  public interface BucketConsumer {
+    void accept(double upperBound, long count);
+  }
+
+  /**
+   * Creates a DoubleHistogramPointData.
+   *
+   * @return a DoubleHistogramPointData.
+   */
+  public static DoubleHistogramPointData create(
+      long startEpochNanos,
+      long epochNanos,
+      Labels labels,
+      double sum,
+      long count,
+      ImmutableDoubleArray boundaries,
+      ImmutableLongArray counts) {
+    return new AutoValue_DoubleHistogramPointData(
+        startEpochNanos, epochNanos, labels, sum, count, boundaries, counts);
+  }
+
+  DoubleHistogramPointData() {}
+
+  /**
+   * The sum of all measurements recorded.
+   *
+   * @return the sum of recorded measurements.
+   */
+  public abstract double getSum();
+
+  /**
+   * The number of measurements taken.
+   *
+   * @return the count of recorded measurements.
+   */
+  public abstract long getCount();
+
+  /**
+   * The bucket boundaries. For a Histogram with N defined boundaries, e.g, [x, y, z]. There are N+1
+   * counts: [-inf, x), [x, y), [y, z), [z, +inf].
+   *
+   * @return the bucket boundaries in increasing order.
+   */
+  public abstract ImmutableDoubleArray getBoundaries();
```

Comment:
We can have a List here, and we can return "unmodifiableList".

Reply:
assume this is the result of marking `ImmutableDoubleArray` non-public so we can't use the class here?

imo, `ImmutableDoubleArray` is a better contract than `unmodifiableList` because the immutable property is guaranteeed at compile-time while the latter could cause a run time exception.

Reply:
I've mark `ImmutableDoubleArray` and `ImmutableLongArray` non-public so that it's only visible in the aggregator package. But this kind of makes the interface of the `DoubleHistogramPointData` dependent on the mutable `List` interface (it's `Collections.unmodifiableList` tho).